# Extracted Document

*Extracted with Ultra-Quality DeepSeek-OCR (300 DPI, Gundam-M mode)*

---


## Page 1

# Activating ESR1 mutations in hormone-resistant metastatic breast cancer

Dan R Robinson \(^{1,2,12}\) , Yi- Mi Wu \(^{1,2,12}\) , Pankaj Vats \(^{1,2}\) , Fengyun Su \(^{1,2}\) , Robert J Lonigro \(^{1,3}\) , Xuhong Cao \(^{1,4}\) , Shanker Kalyana- Sundaram \(^{1,2}\) , Rui Wang \(^{1,2}\) , Yu Ning \(^{1,2}\) , Lynda Hodges \(^{1}\) , Amy Gursky \(^{1,2}\) , Javed Siddiqui \(^{1,2}\) , Scott A Tomlins \(^{1,2}\) , Sameek Roychowdhury \(^{5}\) , Kenneth J Pienta \(^{6}\) , Scott Y Kim \(^{7}\) , J Scott Roberts \(^{8}\) , James M Rae \(^{3,9}\) , Catherine H Van Poznak \(^{9}\) , Daniel F Hayes \(^{9}\) , Rashmi Chugh \(^{9}\) , Lakshmi P Kunju \(^{1,2}\) , Moshe Talpaz \(^{9}\) , Anne F Schott \(^{9}\) & Arul M Chinnaiyan \(^{1- 4,10,11}\)

Breast cancer is the most prevalent cancer in women, and over two- thirds of cases express estrogen receptor- \(\alpha\) (ER- \(\alpha\) , encoded by ESR1). Through a prospective clinical sequencing program for advanced cancers, we enrolled 11 patients with ER- positive metastatic breast cancer. Whole- exome and transcriptome analysis showed that six cases harbored mutations of ESR1 affecting its ligand- binding domain (LBD), all of whom had been treated with anti- estrogens and estrogen deprivation therapies. A survey of The Cancer Genome Atlas (TCGA) identified four endometrial cancers with similar mutations of ESR1. The five new LBD- localized ESR1 mutations identified here (encoding p.Leu536Gln, p.Tyr537Ser, p.Tyr537Cys, p.Tyr537Asn and p.Asp538Gly) were shown to result in constitutive activity and continued responsiveness to anti- estrogen therapies in vitro. Taken together, these studies suggest that activating mutations in ESR1 are a key mechanism in acquired endocrine resistance in breast cancer therapy.

Advances in high- throughput sequencing technologies are beginning to establish a molecular taxonomy for a spectrum of human diseases and has facilitated a move toward 'precision medicine' (refs. 1,2). With regard to oncology, defining the mutational landscape of a patient's tumor will lead to more precise treatment and management of individuals with cancer. Comprehensive clinical sequencing programs for cancer patients have been initiated at a variety of medical centers, including our own \(^{3,4}\) . In addition to the potential for identifying 'actionable' therapeutic targets in cancer patients, these clinical sequencing efforts may also shed light on acquired resistance mechanisms developed against targeted therapies \(^{5- 7}\) .

ER is the primary therapeutic target in breast cancer and is expressed in \(70\%\) of cases \(^{8}\) . Drugs directly antagonizing ER, such as tamoxifen and fulvestrant, are a mainstay of breast cancer treatment; however, approximately \(30\%\) of ER- positive breast cancers exhibit de novo resistance, whereas \(40\%\) acquire resistance to these therapies \(^{9}\) . In addition to anti- estrogen therapies, patients with ER- positive breast cancer are also treated with aromatase inhibitors such as letrozole and exemestane \(^{10}\) . Aromatase inhibitors block the peripheral conversion of androgens into estrogen and, in post- menopausal women, lead to over a \(98\%\) decrease in circulating levels of estrogen. As with anti- estrogens, treatment with aromatase inhibitors results in the development of resistance, but this is presumably due to different mechanisms, as patients with breast cancer who develop resistance to aromatase inhibitors often still respond to anti- estrogen therapies \(^{11}\) . The molecular mechanisms of endocrine resistance in ER- positive breast cancer continues to be an active area of research \(^{12}\) .

Our institutional review board (IRB)- approved clinical sequencing program, called MI- ONCOSEQ (the Michigan Oncology Sequencing Program), enrolls patients with advanced cancer across all histologies \(^{3}\) . Since April 2011, we have enrolled over 200 patients in this program, which involves obtaining a current tumor biopsy with matched normal samples (blood and/or buccal swab). Samples are then subjected to integrative sequencing, which includes whole- exome sequencing of the tumor and matched normal sample, transcriptome sequencing and, as needed, low- pass whole- genome sequencing \(^{3}\) . This combination of DNA and RNA sequencing technologies allows one to be relatively comprehensive with regard to the mutational landscape of coding genes, including analysis of point mutations, indels, amplifications, deletions, gene fusions or translocations, and outlier gene expression profiles. These results are generated within a 5- to 7- week time frame and are presented at an institutional 'precision medicine tumor board' to deliberate upon potentially actionable findings.


## Page 2

![Table](extracted_images/page2_table0.jpg)

Table 1 Clinical sequencing of 11 metastatic ER-positive breast cancer cases   

<table><tr><td>Case</td><td>Age (years)</td><td>ER/PR/ERBB2</td><td>Treatmenta</td><td>Number of SNVs/fusions</td><td>Genetic aberrationb</td></tr><tr><td>MO_1031</td><td>41</td><td>+/+/-</td><td>Tamoxifen, letrozole, fulvestrant</td><td>266/18</td><td>ESR1 (p.Leu536Gln), gene copy gains of FGFR1, FGFR2, CCND1 and GNRHR</td></tr><tr><td>MO_1051</td><td>31</td><td>+/--/-</td><td>Oophorectomy, letrozole, fulvestrant</td><td>248/5</td><td>ESR1 (p.Tyr537Ser), PIK3CA (p.His1047Arg), TP53 (p.Gly199Glu), FGFR2-AFF3 fusion</td></tr><tr><td>MO_1069</td><td>62</td><td>+/+/-</td><td>Tamoxifen, letrozole, fulvestrant</td><td>74/9</td><td>ESR1 (p.Asp538Gly), ARID2 (p.Glu245*), gene copy losses of TP53, BRCA1, RB1, ARID1A and SMARCA4</td></tr><tr><td>MO_1129</td><td>44</td><td>+/+/-</td><td>Tamoxifen, oophorectomy, anastrozole, fulvestrant, exemestane</td><td>32/3</td><td>ESR1 (p.Tyr537Ser), PIK3CA (p.Glu542Lys), gene copy gains of CCND1 and PAK1</td></tr><tr><td>MO_1030</td><td>78</td><td>+/+/-</td><td>Tamoxifen (short), anastrozole, fulvestrant</td><td>26/2</td><td>PIK3CA (p.Glu545Ala), TP53 copy loss</td></tr><tr><td>MO_1068</td><td>65</td><td>+/--/-</td><td>Tamoxifen, anastrozole</td><td>83/10</td><td>PIK3CA (p.His1047Arg), TP53 (p.Glu51*), MSH2 copy loss</td></tr><tr><td>MO_1090</td><td>52</td><td>+/+/-</td><td>Tamoxifen, anastrozole</td><td>28/11</td><td>No significant drivers identified</td></tr><tr><td>MO_1107</td><td>46</td><td>+/+/-</td><td>Tamoxifen, oophorectomy, anastrozole, fulvestrant, exemestane</td><td>63/12</td><td>BRCA1 (c.5385_5386insC), frameshift deletions in TP53, SMARCA4 and NF1</td></tr><tr><td>MO_1167</td><td>60</td><td>+/--/-</td><td>Tamoxifen, letrozole</td><td>47/3</td><td>ESR1 (p.Asp538Gly)</td></tr><tr><td>MO_1185</td><td>58</td><td>+/+/-</td><td>Tamoxifen, letrozole, fulvestrant, exemestane</td><td>88/1</td><td>ESR1 (p.Tyr537Ser), CDH1 (p.Gln641*), NOTCH2 (frameshift deletion), TP53 copy loss</td></tr><tr><td>TP_2004c</td><td>52</td><td>+/--/-</td><td>Tamoxifen (short)</td><td>29/22</td><td>MDM2 gene amplification, gene copy losses of CDKN2A and CDKN2B</td></tr></table>

PR, progesterone receptor; SNVs, single-nucleotide variants. Only anti-estrogen-related treatments are listed. Patients also received chemotherapies, radiation or mastectomy in the interim between diagnosis and MI-ONCOSEQ sequencing. bAmino acid substitutions caused by nonsynonymous somatic mutations are given in parentheses. TP_2004 is male.

As part of the MI- ONCOSEQ program, we enrolled and sequenced 11 patients with metastatic ER- positive breast cancer (Table 1 and Supplementary Table 1). A diverse array of aberrations were identified in individual patients, some of which are potentially actionable, including mutations in PIK3CA \((n = 4)\) , BRCA1 aberrations \((n = 2)\) , FGFR2 aberrations \((n = 2)^{13}\) , NOTCH2 frameshift deletion \((n = 1)\) , cyclin and associated cyclin- dependent kinase aberrations \((n = 3)\) and MDM2 amplification and overexpression \((n = 1)\) . Aberrations were also frequently found in the tumor suppressor TP53 \((n = 6)\) , the DNA mismatch repair gene MSH2 \((n = 1)\) and in epigenetic regulators \((n = 2)\) , including ARID2, ARID1A and SMARCA4, among others. The complete spectra of somatic mutations with associated alterations in expression levels and copy number in the index patients are given in Supplementary Figure 1 and Supplementary Tables 2 and 3. Two of the index patients, MO_1031 and MO_1051, exhibited a high level of mutations consistent with 'signature B' identified in a whole- genome study of mutational processes in breast cancer \(^{14}\) . There were 39 gene fusions identified in the 6 index patients, with 11 encoding in- frame fusion proteins (Supplementary Fig. 2 and Supplementary Tables 4 and 5), including an activating FGFR2- AFF3 fusion \(^{13}\) .

The most notable observation in the mutational landscapes of these treated patients with ER- positive breast cancer was the finding of nonsynonymous mutations in ESR1 affecting the LBD \((n = 6)\) . The six index patients MO_1031, MO_1051, MO_1069, MO_1129, MO_1167 and MO_1185 had mutations encoding p.Leu536Gln, p.Tyr537Ser, p.Asp538Gly, p.Tyr537Ser, p.Asp538Gly and p.Tyr537Ser alterations in the LBD, respectively. The respective mutation in each case was detected by whole- exome sequencing of the tumor relative to the matched normal sample and was corroborated by whole- transcriptome sequencing, as ESR1 was expressed at moderate to high levels (Supplementary Table 2). The clinical histories of the index patients are depicted in timelines in Figure 1. For three of the patients (MO_1051, MO_1069 and MO_1129), we had access to primary diagnostic material and showed that the ESR1 mutations were not present at an earlier stage, indicating that they were acquired after endocrine therapy (Fig. 1 and Supplementary Table 2). Interestingly, all of the index patients were treated with anti- estrogens (tamoxifen and/or fulvestrant)

and aromatase inhibitors (letrozole, anastrozole and/or exemestane). Two of the patients also had an oophorectomy. Comparison of the mutations present in each primary versus post- treatment pair showed a substantial number of shared mutations in both samples of the pair, including activating mutations in PIK3CA in two of the cases. Thus, it is clear that the index cases presented with recurrent disease of the original primary tumor surviving in an estrogen- deprived state and having acquired ESR1 mutations. Of note, neither ESR1 amplifications nor gene fusions were observed in these cases.

The five new LBD alterations of ESR1 identified in this study are depicted in Figure 2. Each occurred in the vicinity of the synthetic alterations of ESR1 that are inverted in response to tamoxifen and involve p.Met543Ala and p.Leu544Ala alterations (Inv- mut- AA2) \(^{15}\) and served as a positive control for our subsequent in vitro studies. We next investigated the occurrence of ESR1 mutations in a range of breast cancer types. Here we took advantage of data from the TCGA Project, which has generated whole- exome sequences for 27 tumor types across at least 4,000 individual samples. As expected, LBD- disrupting mutations of ESR1 were not detected in the 390 ER- positive breast cancers sequenced by TCGA, as these were primary resection samples before hormonal treatment \(^{16}\) , nor did we detect ESR1 mutations in a cohort of 80 triple- negative breast carcinoma transcriptomes (D.R.R., Y.- M.W., X.C., S.K.- S., A.M.C. et al., unpublished data).

As the LBD- disrupting mutations of ESR1 we identified were somatic and were acquired after treatment, we next assessed whether the encoded proteins were dependent on estrogen for activation. We cloned into expression vectors each of the five ESR1 mutants identified in this study (encoding p.Leu536Gln, p.Tyr537Ser, p.Asp538Gly, p.Tyr537Cys and p.Tyr573Asn alterations) and subsequently cotransfected these constructs into HEK293T cells with an estrogen response element (ERE)- luciferase reporter system. We then exposed steroid hormone- deprived cells to \(\beta\) - estradiol for 24 h and assessed ERE reporter levels. Surprisingly, unlike wild- type ESR1, which had little ERE reporter activity in the absence of ligand, all five of the ESR1 mutants had strong constitutive activation of the ERE reporter that was not markedly enhanced with \(\beta\) - estradiol (Fig. 3). This finding suggested that each of the mutations developed in the context of evolution during


## Page 3

![Image](extracted_images/page3_image0.jpg)

*Figure 1: Image content could not be automatically extracted. See cropped image above.*

<center>Figure 1 Clinical timelines for the six index ER-positive metastatic breast cancer patients harboring ESR1 mutations. Shown are patients' histories of clinical treatment from first diagnosis until enrollment in the MI-ONCOSEQ study. Each bar represents the timeframe of treatment. FFPE, formalin fixed, paraffin embedded; CAF, cyclophosphamide, doxorubicin, fluorouracil chemotherapy. </center>

an estrogen- deprived state. Consistent with this idea, a whole- genome sequencing study of 46 patients with ER- positive breast cancer enrolled in 2 aromatase inhibitor trials did not identify any of these ESR1 mutations in the pretreatment samples analyzed17.

Next, we assessed whether anti- estrogen therapies affected the functional activity of the LBD mutants. As effects on inhibition can be influenced by the levels of ectopic ER expression, we performed a dose response study with expression plasmid and selected a dose of \(50 \mathrm{ng}\) for the following experiments18 (Supplementary Fig. 3). As expected, wild- type ESR1 was inhibited in a dose- dependent fashion by the anti- estrogens 4- hydroxytamoxifen, fulvestrant and endoxifen (Fig. 4 and

Supplementary Figs. 4- 6). In addition, the mutant corresponding to the synthetic ESR1 mutation (Inv- mut- AA2) was activated in a dose- dependent fashion by these anti- estrogens (Fig. 4), which has been reported previously15. Interestingly, ESR1 with each of the five LBD alterations identified in this study was inhibited by tamoxifen and fulvestrant in a dose- dependent fashion and did not exhibit the inverted response to anti- estrogens that the synthetic Inv- mut- AA2 mutant did. One could speculate that the corresponding mutations did not arise under selective pressure of anti- estrogen treatment but rather in the context of an estrogen deprivation setting, such as treatment with aromatase inhibitors and/or oophorectomy. The \(\mathrm{IC}_{50}\) (half- maximal


## Page 4

Figure 2 Schematic of ESR1 alterations identified in this study. The structural domains of ESR1 are illustrated on top, including the transcription activation function- 1 domain (AF- 1), the DNA- binding domain (DBD), the hinge domain and the ligand- binding domain (LBD/AF- 2). Altered residues identified in mutants are marked in red, and reference residues are shown in bold in the wild- type sequence. Endometrium p.Tyr537Cys and p.Tyr537Asn are two alterations discovered in endometrial cancer samples from the TCGA study. Inv- mut- AA2 represents a ligand activity inversion mutant of ESR1 that confers inverted responses to anti- estrogen and estrogen. H11, helix 11; H12, helix 12.

![Image](extracted_images/page4_image1.jpg)

*Figure 2: Image content could not be automatically extracted. See cropped image above.*

inhibitory concentration) values for both 4- hydroxy tamoxifen and fulvestrant were two- to fourfold higher for all mutants compared to wild- type ESR1. Fulvestrant exhibited greater maximal inhibition than 4- hydroxy tamoxifen for all the mutants tested (Supplementary Figs. 4 and 5).

The ESR1 alterations identified in this study cluster near the beginning of helix 12 (Fig. 2). Structural studies have demonstrated a key role for the position of helix 12 in the response of the ER to agonists and antagonists19, and Tyr537 has been postulated to form a capping motif contributing to the activity of the receptor20. Specifically, the p.Tyr537Ser mutant has been reported to have higher affinity for estrogen than wild- type ESR1 and interacts with the SRC1 coactivator in the absence of ligand21,22. Several studies using experimental mutagenesis have implicated the same three residues identified here as critical determinants of the transcriptional activity of the receptor21,23,24.

As estrogen therapy has been shown to have a positive effect in treating aromatase inhibitor- resistant advanced breast cancers, we tested the effect of low- to high- dose estrogen on the activity of the mutants in the transient luciferase reporter assays (Supplementary Fig. 7)25,26. The results did not suggest that the effectiveness of this therapy is mediated through direct control of the transcriptional activity of these mutants, if encoded by the responding patients.

Although the primary intent of our broad- based clinical sequencing program is to identify actionable and/or driver mutations in advanced cancers, this study demonstrates how such prospective, real- time sequencing efforts can also shed light on resistance mechanisms that develop against targeted therapies. A number of resistance mechanisms have been suggested to function in the evasion of endocrine treatment, including activation of the mTOR and phosphoinositide 3- kinase (PI3K) pathways, among others9,27. Although the total

![Image](extracted_images/page4_image6.jpg)

*Figure 7: Image content could not be automatically extracted. See cropped image above.*

number of ER- positive breast cancers we have sequenced is modest, we have done so in a comprehensive fashion in terms of delineating mutational landscapes and incorporating both DNA and RNA sequencing. This analysis identified de novo driver mutations and/or potentially acquired mutations in breast cancer such as mutations resulting in PI3K activation, PAK1 amplification and FGFR fusion and amplification, which have been described previously13,28,29. Among potential new mechanisms described, we identified profound focal amplification of MDM2 (which encodes a negative regulator of p53 that is targetable) and copy gains of GNRHR (encoding gonadotropin- releasing hormone receptor), which may be related to past endocrine therapy.

As the LBD- disrupting mutations of ESR1 identified in this study result in constitutive activity, the encoded mutant proteins can function in the absence of ligand and maintain ER signaling. In 1997, an ESR1 mutation affecting the LBD, encoding a p.Tyr537Asn alteration, was detected in a single individual with stage IV metastatic breast cancer who had been treated with diethylstilbestrol, but, since then, this mutation has been considered to be very rare30. With the advent of widespread aromatase inhibitor therapy, we suggest that alteration of the ESR1 LBD is likely a common mechanism of resistance that develops in low- estrogen states. Interestingly, LBD- disrupting mutations of ESR1 were detected somatically in 4 of 373 cases of endometrial cancer31. We speculate that the four TCGA endometrial tumors that harbor LBD- affecting mutations likely came from patients with concurrent breast cancer, as tamoxifen treatment is known to be associated with higher incidence of this tumor type and such patients also often receive estrogen deprivation treatment32.

Our study suggests that it is unlikely that these LBD alterations develop in the context of anti- estrogen treatment, as the mutated ESR1 variants continue to be responsive to direct ER antagonists such as tamoxifen and fulvestrant. This finding is consistent with clinical reports showing that patients that develop resistance to aromatase inhibitors still respond to anti- estrogen treatment11. Although this prospective clinical sequencing study was not designed to characterize a specific cancer type or treatment resistance mechanism,

Figure 3 ESR1 with acquired mutations encodes constitutively active protein. HEK293T cells were cotransfected with an ERE- firefly luciferase reporter plasmid, a plasmid constitutively expressing Renilla luciferase as an internal control and various ESR1 constructs (as illustrated in Fig. 2). Steroid hormone- deprived cells were either untreated or stimulated with \(5\mathrm{nM}\beta\) - estradiol for \(24\mathrm{h}\) . Firefly luciferase activities were normalized using corresponding Renilla luciferase activities for each condition. Fold change in ESR1- driven transcriptional activity was calculated using untreated wild type as a control for each condition. Data shown are the means from triplicate experiments. Amino acid changes in respective ESR1 mutants are indicated. WT, wild- type ESR1. Error bars, s.d.


## Page 5

Figure 4 Acquired ESR1 mutations result in maintained sensitivity to anti- estrogen therapies. HEK293T cells were cotransfected with an ERE- firefly luciferase reporter plasmid, a plasmid constitutively expressing Renilla luciferase and various ESR1 constructs as indicated. (a,b) Steroid hormone- deprived cells were either untreated or treated with increasing doses of the anti- estrogen drugs tamoxifen (a) and fulvestrant (b) in the presence of \(5 \mathrm{nM}\) \(\beta\) - estradiol (E2) for \(24 \mathrm{h}\) . Percentage change in ESR1- driven transcriptional activity was calculated using E2- treated cells as the control for each tested construct. Data shown are the means from triplicate experiments. Error bars, s.d. \(^{*}P < 0.001\) . Red, wild type; blue, clinically identified mutants; orange, synthetic ligand- inversion mutant; green, vector control.

future studies comprising larger cohorts of breast cancer patients with disease that recurs after varied endocrine treatments will more precisely delineate the incidence of this acquired resistance mechanism. The focused nature of these mutations and their role in aromatase inhibitor resistance suggest the possibility of monitoring patients undergoing treatment using circulating tumor DNA methods \(^{33,34}\) . In this manner, treatment could be shifted to head off evolving tumor resistance.

URLs. BLAT, http://genome.ucsc.edu/cgi- bin/hgBlat; ORF Finder, http://www.ncbi.nlm.nih.gov/gorf/gorf.html.

## METHODS

Methods and any associated references are available in the online version of the paper.

Accession codes. Sequence data have been deposited in the database of Genotypes and Phenotypes (dbGaP), which is hosted by the National Center for Biotechnology Information (NCBI), under accessions phs000602. v1. p1 and phs000673. v1. p1.

Note: Any Supplementary Information and Source Data files are available in the online version of the paper.

## ACKNOWLEDGMENTS

The authors thank D. Miller, T. Barrette and D. Gibbs for hardware and database management, K. Giles for assistance with manuscript preparation, physicians M. Wicha, L. Pierce, D. Smith, K. Levin and F. Feng for referring patients, and C. Betts and J. Athanikar for assistance with tumor boards. We also thank the larger MI- ONCOSEQ team, including clinical research coordinator E. Williams, pathologist R. Mehra, genetic counselors J. Everett, S. Gustafson and V. Raymond, and radiologists E. Higgins, E. Caoili and R. Dunnick. This project is supported in part by the Prostate Cancer Foundation with funding for our sequencing infrastructure, the National Cancer Institute Early Detection Research Network (U01 CA111275), National Human Genome Research Institute Clinical Sequencing Exploratory Research (CSER) Consortium (1UM1HG006508), US Department of Defense contract W81XWH- 12- 1- 0080 and a Department of Defense Era of Hope Scholar Award. A.M.C. is also supported by the Alfred A. Taubman Institute, the American Cancer Society, the Howard Hughes Medical Institute and a Doris Duke Charitable Foundation Clinical Scientist Award.

## AUTHOR CONTRIBUTIONS

D.R.R., Y.- M.W. and A.M.C. conceived the experiments. D.R.R., Y.- M.W., X.C., R.W., F.S. and Y.N. performed exome and transcriptome sequencing. P.V., R.J.L.,

![Image](extracted_images/page5_image11.jpg)

*Figure 12: Image content could not be automatically extracted. See cropped image above.*

S.K.- S. and D.R.R. carried out bioinformatics analysis of high- throughput sequencing data for somatic mutation, copy number and tumor content determination and performed gene expression and gene fusion analysis. D.R.R., Y.- M.W. and F.S. generated ESR1 constructs and carried out in vitro experiments. L.H. coordinated patients for clinical research. J.S. and A.G. collected and processed clinical tissue samples for next- generation sequencing. L.P.K. and S.A.T. provided pathology review. J.M.R. provided experimental analysis. C.H.V.P., D.F.H., R.C. and A.F.S. enrolled patients and provided clinical data and consultation at tumor boards. D.R.R., X.C., Y.- M.W., P.V., R.J.L., S.K.- S., S.Y.K., J.S.R., S.R., M.T., K.J.P. and A.M.C. developed the integrated clinical sequencing protocol. D.R.R., Y.- M.W. and A.M.C. prepared the manuscript, which was reviewed by all authors.

## COMPETING FINANCIAL INTERESTS

The authors declare no competing financial interests.

Reprints and permissions information is available online at http://www.nature.com/reprints/index.html.

1. Chin, L., Andersen, J.N. & Futreal, P.A. Cancer genomics: from discovery science to personalized medicine. Nat. Med. 17, 297-303 (2011).  
2. Meyerson, M., Gabriel, S. & Getz, G. Advances in understanding cancer genomes through second-generation sequencing. Nat. Rev. Genet. 11, 685-696 (2010).  
3. Roychowdhury, S. et al. Personalized oncology through integrative high-throughput sequencing: a pilot study. Sci. Transl. Med. 3, 111ra121 (2011).  
4. Welch, J.S. et al. Use of whole-genome sequencing to diagnose a cryptic fusion oncogene. J. Am. Med. Assoc. 305, 1577-1584 (2011).  
5. Gorre, M.E. et al. Clinical resistance to STI-571 cancer therapy caused by BCR-ABL gene mutation or amplification. Science 293, 876-880 (2001).  
6. Korpal, M. et al. An F876L mutation in androgen receptor confers genetic and phenotypic resistance to MDV3100 (enzalutamide). Cancer Discov. 3, 1030-1043 (2013).  
7. Joseph, J.D. et al. A clinically relevant androgen receptor mutation confers resistance to 2nd generation anti-androgens enzalutamide and ARN-509. Cancer Discov. 3, 1020-1029 (2013).  
8. Ariazi, E.A., Ariazi, J.L., Cordera, F. & Jordan, V.C. Estrogen receptors as therapeutic targets in breast cancer. Curr. Top. Med. Chem. 6, 181-202 (2006).  
9. Riggins, R.B., Schrecengost, R.S., Guerrero, M.S. & Bouton, A.H. Pathways to tamoxifen resistance. Cancer Lett. 256, 1-24 (2007).


## Page 6

10. LÃ¸nning, P.E. & Eikesdal, H.P. Aromatase inhibition 2013: clinical state of the art and questions that remain to be solved. Endocr. Relat. Cancer 20, R183-R201 (2013).  
11. Ingle, J.N. et al. Fluvestrant in women with advanced breast cancer after progression on prior aromatase inhibitor therapy: North Central Cancer Treatment Group Trial N0032. J. Clin. Oncol. 24, 1052-1056 (2006).  
12. Osborne, C.K. & Schiff, R. Mechanisms of endocrine resistance in breast cancer. Annu. Rev. Med. 62, 233-247 (2011).  
13. Wu, Y.M. et al. Identification of targetable FGFR gene fusions in diverse cancers. Cancer Discov. 3, 636-647 (2013).  
14. Nik-Zainal, S. et al. Mutational processes molding the genomes of 21 breast cancers. Cell 149, 979-993 (2012).  
15. Feil, R., Wagner, J., Metzger, D. & Chambon, P. Regulation of Cre recombinase activity by mutated estrogen receptor ligand-binding domains. Biochem. Biophys. Res. Commun. 237, 752-757 (1997).  
16. TCGA. Comprehensive molecular portraits of human breast tumours. Nature 490, 61-70 (2012).  
17. Ellis, M.J. et al. Whole-genome analysis informs breast cancer response to aromatase inhibition. Nature 486, 353-360 (2012).  
18. Huang, H.J., Norris, J.D. & McDonnell, D.P. Identification of a negative regulatory surface within estrogen receptor \(\alpha\) provides evidence in support of a role for corepressors in regulating cellular responses to agonists and antagonists. Mol. Endocrinol. 16, 1778-1792 (2002).  
19. Shiau, A.K. et al. The structural basis of estrogen receptor/coactivator recognition and the antagonism of this interaction by tamoxifen. Cell 95, 927-937 (1998).  
20. Skafar, D.F. Formation of a powerful capping motif corresponding to start of "helix 12" in agonist-bound estrogen receptor-\(\alpha\) contributes to increased constitutive activity of the protein. Cell Biochem. Biophys. 33, 53-62 (2000).  
21. Carlson, K.E., Choi, I., Gee, A., Katzenellenbogen, B.S. & Katzenellenbogen, J.A. Altered ligand binding properties and enhanced stability of a constitutively active estrogen receptor: evidence that an open pocket conformation is required for ligand interaction. Biochemistry 36, 14897-14905 (1997).

22. Weis, K.E., Ekena, K., Thomas, J.A., Lazennec, G. & Katzenellenbogen, B.S. Constitutively active human estrogen receptors containing amino acid substitutions for tyrosine 537 in the receptor protein. Mol. Endocrinol. 10, 1388-1398 (1996).  
23. Pearce, S.T., Liu, H. & Jordan, V.C. Modulation of estrogen receptor \(\alpha\) function and stability by tamoxifen and a critical amino acid (Asp-538) in helix 12. J. Biol. Chem. 278, 7630-7638 (2003).  
24. Zhao, C. et al. Mutation of Leu-536 in human estrogen receptor-\(\alpha\) alters the coupling between ligand binding, transcription activation, and receptor conformation. J. Biol. Chem. 278, 27278-27286 (2003).  
25. Ellis, M.J. et al. Lower-dose vs high-dose oral estradiol therapy of hormone receptor-positive, aromatase inhibitor-resistant advanced breast cancer: a phase 2 randomized study. J. Am. Med. Assoc. 302, 774-780 (2009).  
26. Swaby, R.F. & Jordan, V.C. Low-dose estrogen therapy to reverse acquired antihormonal resistance in the treatment of breast cancer. Clin. Breast Cancer 8, 124-133 (2008).  
27. Sokolosky, M.L. et al. Involvement of Akt-1 and mTOR in sensitivity of breast cancer to targeted therapy. Oncotarget 2, 538-550 (2011).  
28. Kan, Z. et al. Diverse somatic mutation patterns and pathway alterations in human cancers. Nature 466, 869-873 (2010).  
29. Shrestha, Y. et al. PAK1 is a breast cancer oncogene that coordinately activates MAPK and MET signaling. Oncogene 31, 3397-3408 (2012).  
30. Barone, I., Brusco, L. & Fuqua, S.A. Estrogen receptor mutations and changes in downstream gene expression and signaling. Clin. Cancer Res. 16, 2702-2708 (2010).  
31. Kandoth, C. et al. Integrated genomic characterization of endometrial carcinoma. Nature 497, 67-73 (2013).  
32. Fisher, B. et al. Endometrial cancer in tamoxifen-treated breast cancer patients: findings from the National Surgical Adjuvant Breast and Bowel Project (NSABP) B-14. J. Natl. Cancer Inst. 86, 527-537 (1994).  
33. Dawson, S.J. et al. Analysis of circulating tumor DNA to monitor metastatic breast cancer. N. Engl. J. Med. 368, 1199-1209 (2013).  
34. Diehl, F. et al. Circulating mutant DNA to assess tumor dynamics. Nat. Med. 14, 985-990 (2008).


## Page 7

## ONLINE METHODS

Clinical study and specimen collection. Sequencing of clinical samples was performed under IRB- approved studies at the University of Michigan. Patients were enrolled and consented for integrative tumor sequencing in MI- ONCOSEQ (Michigan Oncology Sequencing Protocol, HUM00046018). Medically qualified patients 18 years or older with advanced or refractory cancer were eligible for the study. Informed consent detailed the risks of integrative sequencing and includes up- front genetic counseling. Informed consent was obtained from all subjects included in this study. Biopsies were arranged for safely accessible tumor sites. Needle biopsies were snap frozen in OCT (Optimal Cutting Temperature) compound, and a longitudinal section was cut. Frozen sections stained with hematoxylin and eosin were reviewed by pathologists to identify cores with the highest tumor content. Remaining portions of each needle biopsy core were retained for nucleic acid extraction.

Extraction of DNA and RNA. Genomic DNA from frozen needle biopsies and blood was isolated using the Qiagen DNeasy Blood and Tissue kit, according to the manufacturer's instructions. Total RNA was extracted from frozen needle biopsies using the Qiazol reagent with disruption using a \(5\mathrm{- mm}\) bead on a Tissuelyser II (Qiagen) and was purified using a miRNeasy kit (Qiagen) with DNase I digestion, according to the manufacturer's instructions. RNA integrity was verified on an Agilent 2100 Bioanalyzer using RNA Nano reagents (Agilent Technologies).

Preparation of next- generation sequencing libraries. Transcriptome libraries were prepared using \(1 - 2\mu \mathrm{g}\) of total RNA. Polyadenylated RNA was isolated using Sera- Mag oligo(dT) beads (ThermoScientific) and fragmented with the Ambion Fragmentation Reagents kit. cDNA synthesis, end repair, A- base addition and ligation of the Illumina indexed adaptors were performed according to Illumina's TrueSeq RNA protocol. Libraries were selected for DNA fragments of \(250 - 300\mathrm{bp}\) in size on a \(3\%\) Nusieve 3:1 agarose gel (Lonza), recovered using QIAEX II gel- extraction reagents (Qiagen) and PCR amplified using Phusion DNA polymerase (New England BioLabs). Amplified libraries were purified using AMPure XP beads (Beckman Coulter). Library quality was measured on an Agilent 2100 Bioanalyzer by product size and concentration. Paired- end libraries were sequenced with the Illumina HiSeq 2000 platform \(2\times 100\) - nucleotide read length). Reads that passed the chastity filter of Illumina BaseCall software were used for subsequent analysis.

Exome libraries of matched pairs of tumor and normal genomic DNA were generated using the Illumina TrueSeq DNA Sample Prep kit, following the manufacturer's instructions. In brief, \(1 - 3\mu \mathrm{g}\) of each genomic DNA sample was sheared using a Covaris S2 to a peak target size of \(250\mathrm{bp}\) . Fragmented DNA was concentrated using AMPure XP beads, and end repair, A- base addition and ligation of Illumina indexed adaptors were performed. Adaptor- ligated libraries were electrophoresed on \(3\%\) Nusieve agarose gels, and fragments of \(300 - 350\mathrm{bp}\) were recovered using QIAEX II gel- extraction reagents. Recovered DNA was amplified using Illumina index primers for eight cycles and purified using AMPure XP beads, and DNA concentration was determined using a Nanodrop spectrophotometer. Libraries \((1\mu \mathrm{g})\) were hybridized to the Agilent SureSelect Human All Exon v4 chip at \(65^{\circ}\mathrm{C}\) for \(60\mathrm{h}\) , following the manufacturer's protocol (Agilent Technologies). Targeted exon fragments were captured on Dynal M- 280 streptavidin beads (Invitrogen) and enriched by amplification with the Illumina index primers for nine additional PCR cycles. PCR products were purified with AMPure XP beads and analyzed for quality and quantity using an Agilent 2100 Bioanalyzer and DNA 1000 reagents.

We used the publicly available software FastQC to assess sequencing quality. For each lane, we examined per- base quality scores across the length of the reads. Lanes were deemed passing if the per- base quality score box plot indicated that \(>85\%\) of the reads had \(>Q20\) for bases 1- 100. In addition to raw sequence quality, we also assessed alignment quality using the Picard package. This allows monitoring of duplication rates and chimeric reads that may result from ligation artifacts, crucial statistics for interpreting the results of copy number and structural variant analysis.

Gene fusion detection. Paired- end transcriptome sequencing reads were aligned to the human reference genome (GRCh37/hg19) using an RNA sequencing (RNA- seq) spliced read mapper Tophat2 (ref. 35) (Tophat 2.0.4) with the

- fusion-search' option turned on to detect potential gene fusion transcripts. In the initial process, Tophat2 internally deploys an ultrafast short-read alignment tool, Bowtie (Version 0.12.8), to map the transcriptome data. Potential false-positive fusion candidates were filtered out using the 'Tophat-Post-Fusion' module. Further, fusion candidates were manually examined for annotation and ligation artifacts. Junction reads supporting the fusion candidates were realigned using the BLAT alignment tool to confirm fusion breakpoints. Full-length sequence of each fusion gene was constructed on the basis of supporting junction reads and evaluated for potential ORFs using an ORF Finder. For gene fusions with robust ORFs, the amino acid sequences of the fused proteins were explored using the Simple Modular Architecture Research Tool (SMART) to examine the gain or loss of known functional domains in the fusion proteins.

Gene expression. BAM 'accepted_hits.bam' files, which were generated by the Tophat mapping module, were used to quantify the expression data through Cufflinks36 (Version 2.0.2), an isoform assembly and RNA- seq quantification package. The structural features of 56,369 transcripts from the Ensembl resource (Ensembl 66) were used as an annotation reference to quantify the expression of individual transcripts and isoforms. The 'Max Bundle Length' parameter was set to '10000000', and 'multi- read- correct' was flagged on to perform an initial estimation procedure to more accurately weight reads mapping to multiple locations in the genome.

Mutation analysis. Whole- exome sequencing was performed on an Illumina HiSeq 2000 or HiSeq 2500 instrument in paired- end mode, and primary base call files were converted into FASTQ sequence files using the bcl2fastq converter tool bcl2fastq- 1.8.4 in the CASAVA 1.8 pipeline. FASTQ sequence files were then processed through an in- house pipeline constructed for whole- exome sequence analyses of paired cancer and normal genomes. Sequencing reads were aligned to reference genome build hg19 (GRCh37) using Novoaigin multithreaded (Version 2.08.02, Novocraft) and converted into BAM files using SAMtools (Version 0.1.18)37. Sorting and indexing of BAM files used Novosort threaded (Version 1.00.01), and duplicate reads were removed using Picard (Version 1.74). Mutation analysis was performed using VarScan2 algorithms (Version 2.3.2)38 with the pileup files created by SAMtools mipleup for tumor and matched normal samples, simultaneously performing pairwise comparisons of base call and normalized sequence depth at each position. For SNV detection, filtering parameters including coverage, variant read support, variant frequency, \(P\) value, base quality, the presence of homopolymers and strandness were applied. For indel analysis, Pindel (Version 0.2.4) was used on tumor and matched normal samples, and indels common to both samples were classified as germline, whereas indels present in tumor but not in normal samples were classified as somatic. Finally, a list of candidate indels as well as of somatic and/or germline mutations was generated by excluding synonymous SNVs. ANNOVAR39 was used to functionally annotate the detected genetic variants, and positions are based on Ensembl 66 transcript sequences.

Tumor content for each tumor exome library was estimated from the sequence data by fitting a binomial mixture model with two components to the set of most likely SNV candidates from two- copy genomic regions. The set of candidates used for estimation consisted of coding variants that (i) were supported by at least 3 variant fragments in the tumor sample, (ii) were not supported by variant fragments in the matched benign sample, with at least 16 fragments of coverage, (iii) were not present in dbSNP, (iv) were within a targeted exon or within 100 bp of a targeted exon, (v) were not in homopolymer runs of 4 or more bases and (vi) exhibited no evidence of amplification or deletion. To filter out regions of possible amplification or deletion, we used exon coverage ratios to infer copy number changes, as described below. Resulting SNV candidates were not used for the estimation of tumor content if the segmented log ratio exceeded 0.2 in absolute value. Candidates on the Y chromosome were also eliminated because they were unlikely to exist in two- copy genomic regions. Using this set of candidates, we fit a binomial mixture model with two components using the R package flexmix, version 2.3- 8. One component consisted of SNV candidates with very low variant fractions, presumably resulting from recurrent sequencing errors and other artifacts. The other component, consisting of the set of likely true SNVs, was informative of tumor content in the tumor sample. Specifically,


## Page 8

under the assumption that most or all of the observed SNV candidates in this component are heterozygous SNVs, we expect the estimated binomial proportion of this component to represent one- half of the proportion of tumor cells in the sample. Thus, the estimated binomial proportion obtained from the mixture model was doubled to obtain an estimate of tumor content.

Copy number aberrations were quantified and reported for each gene as the segmented, normalized, \(\log_2\) - transformed exon coverage ratio between each tumor sample and its matched normal sample<sup>40</sup>. To account for observed associations between coverage ratios and variation in GC content across the genome, lowess normalization was used to correct per- exon coverage ratios before segmentation analysis. Specifically, mean GC percentage was computed for each targeted region, and a lowess curve was fit to the scatterplot of \(\log_2\) coverage ratios versus mean GC content across the targeted exome using the lowess function in R (version 2.13.1) with smoothing parameter \(f = 0.05\) .

Partially redundant sequencing of areas of the genome affords the ability for cross- validation of findings. We cross- validated exome- based point mutation calls by manually examining the genomic and transcriptomic reads covering the mutation using the UCSC Genome Browser. Likewise, gene fusion calls from the transcriptome data can be further supported by structural variant detection in the genomic sequence data, as well as by copy number information derived from genome and exome sequencing.

Chemicals and reagents. \(\beta\) - estradiol, (Z)- 4- hydroxy tamoxifen, (E/Z)- endoxifen hydrochloride hydrate and fulvestrant were purchased from Sigma- Aldrich.

Plasmids and cloning. cDNA for wild- type ESR1 was PCR amplified from a breast cell line MCF7 (ATCC) with the introduction of a sequence encoding an N- terminal Flag tag. cDNAs encoding the relevant mutations of ESR1 were generated by site- directed mutagenesis (QuikChange, Agilent Technologies), and full- length constructs were fully sequenced. All ESR1 variants were placed in the lentiviral vector pCDH (System Biosciences) for eukaryotic expression.

ERE- luciferase reporter assays. For cell transfection experiments, HEK293T cells (ATCC) were plated at a density of \(1 - 2 \times 10^5\) cells per well (24- well plates) in phenol red- free DMEM containing \(10\%\) FBS and antibiotics. Once cells attached, the medium was replaced with DMEM containing \(10\%\) charcoal/ dextran- treated FBS (HyClone), and cells were cultured overnight. The next day, cells were transiently cotransfected with ESR1 expression plasmid (50 ng/well) and luciferase reporter constructs (25 ng/well; SABiosciences) using FuGene 6 reagent (Promega). The ER- responsive luciferase plasmid encoding the firefly luciferase reporter gene is driven by a minimal CMV promoter and tandem repeats of the estrogen transcriptional response element (ERE). A second plasmid constitutively expressing Renilla luciferase served as an internal control for normalizing transfection efficiencies (Cignal ERE Reporter, SABiosciences). After transfection for \(18 \text{h}\) , cells were serum starved for a few hours before treatment with \(\beta\) - estradiol or anti- estrogen drugs. Cells were harvested \(18 \text{h}\) after treatment, and luciferase activity was measured using the Dual- Luciferase Reporter Assay System (Promega). IC\(_{50}\) values were computed using the GraphPad Prism application to fit a four- parameter dose response curve.

35. Kim, D. & Salzberg, S.L. TopHat-Fusion: an algorithm for discovery of novel fusion transcripts. Genome Biol. 12, R72 (2011).  
36. Trapnell, C. et al. Differential gene and transcript expression analysis of RNA-seq experiments with TopHat and Cufflinks. Nat. Protoc. 7, 562-578 (2012).  
37. Li, H. et al. The Sequence Alignment/Map format and SAMtools. Bioinformatics 25, 2078-2079 (2009).  
38. Koboldt, D.C. et al. VarScan 2: somatic mutation and copy number alteration discovery in cancer by exome sequencing. Genome Res. 22, 568-576 (2012).  
39. Wang, K., Li, M. & Hakonarson, H. ANNOVAR: functional annotation of genetic variants from high-throughput sequencing data. Nucleic Acids Res. 38, e164 (2010).  
40. Lonigro, R.J. et al. Detection of somatic copy number alterations in cancer using targeted exome capture sequencing. Neoplasia 13, 1019-1025 (2011).

